doc/reference/reference_rock/vshard/vshard_storage.rst

```diff
@@ -310,6 +311,59 @@ Storage public API
             ck_constraint: []
         ...
 
+..  _storage_api-on_bucket_event:
+
+..  function:: vshard.storage.on_bucket_event([trigger-function[, old-trigger-function]])
+
+    Since vshard v.0.1.22. Define a trigger for execution when the data from
+    the user spaces is changed (deleted or inserted) due to the rebalancing
+    process. It's invoked on every data batch changed.
+
+    :param trigger-function: function which will become the trigger function.
+    :param old-trigger-function: existing trigger function which will
+                                 be replaced by trigger-function.
+
+    :return: nil or function pointer
+
+    The trigger-function can have up to three parameters:
+
+      * (string) event_type. In order to distinguish event you could
+        compare this argument with supported event types: ``bucket_data_recv_txn``
+        and ``bucket_data_gc_txn``;
+      * (unsigned) bucket_id;
+      * (table) data. E.g. includes an array of all spaces (data.spaces),
+        affected by a transaction, in which trigger-function is executed.
```

Comment:
```suggestion
    :param function trigger-function: function which will become the trigger function.
    :param function old-trigger-function: existing trigger function which will
                                 be replaced by trigger-function.

    :return: nil or function pointer
    
        The ``trigger-function`` can have up to three parameters:

      * ``event_type``(string) -- in order to distinguish event, you can
         compare this argument with the supported event types, ``bucket_data_recv_txn``
         and ``bucket_data_gc_txn``.
      * ``bucket_id`` (unsigned) -- bucket id.
      * ``data`` (table) -- it may include an array of all spaces (``data.spaces``)
        affected by a transaction in which trigger-function is executed.
```

Reply:
Applied with slight difference: `data` description seems to be too vague. Let's just update it, when new info appears in `data` and state for now, what is really inside it.

Reply:
The rendering is strange if spaces are added in 328 line. I'll omit it, if you don't mind